MechJeb2/MechJebModuleStagingController.cs

```diff
@@ -94,23 +94,27 @@ public override void OnUpdate()
             if (HasStayingChutes(StageManager.CurrentStage - 1, vessel))
                 return;
 
-            //only fire decouplers to drop deactivated engines or tanks or fairings
-            bool firesDecoupler = InverseStageFiresDecoupler(StageManager.CurrentStage - 1, vessel);
-            if (firesDecoupler && !(InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel) || HasFairing(StageManager.CurrentStage - 1, vessel)))
-                return;
-
-            //only decouple fairings if the dynamic pressure and altitude conditions are respected
-            if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
-                HasFairing(StageManager.CurrentStage - 1, vessel))
-                return;
+            //Always drop deactivated engines or tanks
+            if (!InverseStageDecouplesDeactivatedEngineOrTank(StageManager.CurrentStage - 1, vessel))
+            {
+                //only decouple fairings if the dynamic pressure and altitude conditions are respected
+                if ((core.vesselState.dynamicPressure > fairingMaxDynamicPressure || core.vesselState.altitudeASL < fairingMinAltitude) &&
+                    HasFairing(StageManager.CurrentStage - 1, vessel))
+                    return;
+
+                //only release launch clamps if we're at nearly full thrust
+                if (vesselState.thrustCurrent / vesselState.thrustAvailable < 1/1.2 &&

```

Comment:
One of my testing rockets for the PEG stuff has a single F-1 booster stage with about that much SLT and then 4-6 RL-10s that only produce about 0.25-0.35 TWR in the upper stage.  The actual Saturn-V i think was something along those lines.

The SLT clamp also has the added feature that if TestFlight hits you with a failure on the pad, then your clamps wouldn't release if you never made it past 1.00 -- although at some point you'll probably want a way to walk through the engines and determine that they're all green and not Agathorn'd before releasing the clamps.

Ah, and SRBs...   that seems like you'd want to stage the liquid engines, spool them up, then release the SRBs (and i think those still have /some/ spool up time?) and then release the clamps after SLT > 1.00?  because even with SRBs you never want to release clamps with SLT < 1.00

I'm wondering if the pre-stage delay is such a good idea with launch clamp staging?  Maybe you could short-circuit the setting in this case and set it to zero?  Otherwise it seems like we're gonna burn up people's SRBs on the pad or risk dropping them into the pad depending on how we push things around...